src/python/pants/engine/internals/graph_test.py

```diff
@@ -476,46 +432,20 @@ def test_filesystem_specs_no_owner(self) -> None:
 
     def test_resolve_addresses(self) -> None:
         """This tests that we correctly merge addresses resolved from address specs with those
-        resolved from filesystem specs.
-
-        Some important edge cases:
-        - If a filesystem spec resulted in a normal target, and that target is already in the
-          address specs, then we should deduplicate to only use the target one time.
```

Comment:
This case is no longer possible. Filesystem specs will always result in generated subtargets, no matter what.

(`--changed-since` may still result in original base targets thanks to deleted files. But we'll then pass those as `AddressSpecs`. So, it is true even with `--changed-since` that `FilesystemSpecs` will always have generated subtargets.)